### PR TITLE
podman: skip /sys/fs/cgroup/systemd if not present

### PR DESCRIPTION
**Pull request by giuseppe**
skip adding the /sys/fs/cgroup/systemd bind mount if it is not already present on the host.

[NO NEW TESTS NEEDED] requires a system without systemd.

Closes: https://github.com/containers/podman/issues/15647

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Do not attempt to bind mount `/sys/fs/cgroup/systemd` on cgroup v1 if the mount doesn't already exist on the host
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -266,9 +266,15 @@ func (c *Container) setupSystemd(mounts []spec.Mount, g generate.Generator) erro
 		g.AddMount(systemdMnt)
 	} else {
 		mountOptions := []string{"bind", "rprivate"}
+		skipMount := false
 
 		var statfs unix.Statfs_t
 		if err := unix.Statfs("/sys/fs/cgroup/systemd", &statfs); err != nil {
+			if os.IsNotExist(err) {
```
Comment (Luap99) [inline libpod/container_internal_linux.go:273]: please use errors.Is

Comment (rhatdan) [reply libpod/container_internal_linux.go:273]: I always forget this...

Comment (rhatdan) [reply libpod/container_internal_linux.go:273]: Too bad. I wonder if there is linter for this?

**Comment by mheon** [comment]
Code LGTM once the comment is addressed

File: libpod/container_internal_linux.go (hunk 1 above)
Comment (giuseppe) [reply libpod/container_internal_linux.go]: comment addressed and pushed a new version

**Pull request merged**